### PR TITLE
Fix issue 14145 - opDispatch not considered when function body doesn'…

### PR DESCRIPTION
**Pull request by thewilsonator**
…t compile

### CODE REVIEW

File: test/fail_compilation/diag14145.d [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/diag14145.d(15): Error: no property `i` for type `diag14145.main.Capture!(i)`
+fail_compilation/diag14145.d(15):        potentially malformed `opDispatch`. Use an explicit instansiation to get a better error message
```
Comment (thewilsonator) [inline test/fail_compilation/diag14145.d:5]: I know this is not ideal, but its better than nothing. all ideas welcome

**Comment by Imperatorn** [comment]
> …t compile

Should this target stable? 🤔

**Comment by RazvanN7** [comment]
`opDispatch` has the smallest priority, if it was chosen then we might as well print all error messages. This seems like the most straightforward solution, but I realize that it might not be so easy to implement.

**Comment by thewilsonator** [comment]
Theres a bunch of template rewriting done [here](https://github.com/dlang/dmd/blob/58653722e2be7abe61517505c06edf7680c933d1/src/dmd/typesem.d#L3602-L3629) that looks promising but I dunno.

**Comment by thewilsonator** [comment]
> Should this target stable?

Well it probably would do much harm, but the issue is 7 1/2 years old so, shrug?

File: test/fail_compilation/ice10938.d [hunk 2]
```diff
@@ -1,7 +1,8 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/ice10938.d(12): Error: no property `opts` for type `ice10938.C`
+fail_compilation/ice10938.d(13): Error: no property `opts` for type `ice10938.C`
+fail_compilation/ice10938.d(13):        potentially malformed `opDispatch`. Use an explicit instansiation to get a better error message
```
Comment (Imperatorn) [inline test/fail_compilation/ice10938.d:5]: Typo

Instantiation, not Instansiation

**Review by Imperatorn — CHANGES_REQUESTED** [review]
Typo

File: test/fail_compilation/ice10938.d (hunk 2 above)
Comment (thewilsonator) [reply test/fail_compilation/ice10938.d]: fixed

**Comment by adamdruppe** [comment]
This is better than it was, but I wouldn't call it a fix since the compiler should actually just tell you.

Like I think this should be merged immediately, just still I'd probably reopen the issue anyway....

**Comment by thewilsonator** [comment]
Yeah this obviously isn't ideal, I just can't figure out how to synthesise the failed template to recreate the error. If somebody really cares enough to fix it properly then by all means.

**Comment by thewilsonator** [comment]
adding 72h no objections -> merge

**Comment by thewilsonator** [comment]
bugger that I need this already

**Pull request merged**